### PR TITLE
Add crude README to Cargo.toml

### PR DESCRIPTION
**Pull request by phimuemue**
Addressing https://github.com/rust-itertools/itertools/issues/559, this adds our README to Cargo.toml.

Instead of creating a separate one, I suggest simply redirecting to the existing one and update it if needed.

### CODE REVIEW

**Comment by jswrenn** [comment]
AFAIK, crates.io _only_ interprets the file at the `readme` key as markdown (and ours is reStructuredText), so this change would render very poorly.

We'd be better off translating the existing README.rst into markdown, I think.

**Comment by phimuemue** [comment]
I converted it to markdown, and I think they look reasonably similar. Here's the comparable files:

* reStructuredText: https://github.com/rust-itertools/itertools/blob/master/README.rst
* markdown: https://github.com/phimuemue/rust-itertools/blob/readme/README.md

File: README.md [hunk 1]
```diff
@@ -1,50 +1,39 @@
...
-
-|build_status|_ |crates|_
+Please read the [API documentation here](https://docs.rs/itertools/)
 
-.. |build_status| image:: https://travis-ci.org/rust-itertools/itertools.svg?branch=master
-.. _build_status: https://travis-ci.org/rust-itertools/itertools
-
-.. |crates| image:: https://meritbadge.herokuapp.com/itertools
-.. _crates: https://crates.io/crates/itertools
+[![build_status](https://travis-ci.org/rust-itertools/itertools.svg?branch=master)](https://travis-ci.org/rust-itertools/itertools)
```
Comment (atouchet) [inline README.md:7]: This should probably be updated to use GitHub Actions instead of Travis CI.

Comment (phimuemue) [reply README.md:7]: @atouchet Could you possibly review these patches and confirm that the new README.md resembles what we had previously in README.rst? If so, I'd merge it after your review.

Comment (atouchet) [reply README.md:7]: Sorry for missing this earlier. The changes look mostly good to me and I can submit some suggestions as a followup.

**Pull request merged**